gradle/lib/dependabot/gradle/file_updater.rb

```diff
@@ -12,6 +12,17 @@ module Gradle
     class FileUpdater < Dependabot::FileUpdaters::Base
       extend T::Sig
 
+      class BuildFileNotFound < StandardError
+        def initialize(message:, error_context:)
+          super(message)
+          @error_context = error_context
+        end
+
+        def sentry_context
+          { extra: @error_context }
+        end
+      end
```

Comment:
@abdulapopoola is right, this shouldn't be a one-off error... this type of problem is actually a common class of errors across all our package managers where we can't find the dependency file.

Our well known errors are here: https://github.com/dependabot/dependabot-core/blob/ae1e814b35ce06d9d36c210040182250ba22a05a/common/lib/dependabot/errors.rb

It's been a minute since I looked at this, but a quick skim shows separate sections for File Fetcher, File Parser, and File Updater... given this is in the file updater, I suspect you want this section: https://github.com/dependabot/dependabot-core/blob/ae1e814b35ce06d9d36c210040182250ba22a05a/common/lib/dependabot/errors.rb#L162

I don't know if you want `DependencyFileNotResolvable` or `GitDependenciesNotReachable` or `GitDependencyReferenceNotFound`... 

I suggest grep'ing the codebase for other examples of where these exceptions are used to understand which might be the most appropriate.

(I actually wonder if `GitDependencyReferenceNotFound` and `GitDependenciesNotReachable` should be collapsed, but that's a separate conversation).

Reply:
DependencyFileNotResolvable seems a good error class to me